packages/flutter/test/material/dialog_test.dart

```diff
@@ -11,7 +11,7 @@ import 'package:matcher/matcher.dart';
 
 import '../widgets/semantics_tester.dart';
 
-MaterialApp _appWithAlertDialog(WidgetTester tester, AlertDialog dialog, { ThemeData theme }) {
+MaterialApp _buildAppWithDialog(Widget dialog, ThemeData theme) {
```

Comment:
You could just keep this as-is and change the function signature to accept any kind of Widget instead of AlertDialog here. Then you could remove the `_appWithAlertDialog` and `_appWithSimpleDialog` functions below
```suggestion
MaterialApp _buildAppWithDialog(WidgetTester tester, Widget dialog, {ThemeData theme}) {
```

Reply:
Cool. I thought about it but didn't want to change too much. >.<

Can I remove the tester parameter too since it's not used?

Reply:
You could also make it a separate PR to avoid cluttering this one up, but I think it's up to you!

Reply:
As I would need to rename the function because `_appWithAlertDialog` with any widget wouldn't be nice, I've decided to remove the parameter as well. Let me know if all those one-liner changes impacts too much on the diff.